### PR TITLE
Log failed http request url

### PR DESCRIPTION
**Pull request by tobrun**
closes https://github.com/mapbox/mapbox-gl-native/issues/10825.

cc @danswick

### CODE REVIEW

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/http/HTTPRequest.java [hunk 1]
```diff
@@ -180,6 +181,9 @@ private void onFailure(Exception e) {
         // PERMANENT_ERROR
         Timber.w("Request failed due to a permanent error: %s", errorMessage);
       }
+      if (logRequestUrl) {
+        Timber.v(call.request().url().toString());
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/http/HTTPRequest.java]: Can we give some more context to this log and maybe make it a `Timber.w()`? Alternatively, we could include it in above logs with simplified `if`.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/http/HTTPRequest.java [hunk 2]
```diff
@@ -217,4 +221,8 @@ private String getApplicationIdentifier() {
   static void enableLog(boolean enabled) {
     logEnabled = enabled;
   }
+
+  static void enablePrintRequestUrlOnFaillure(boolean enabled) {
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/http/HTTPRequest.java]: "faillure" - typo ;)

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/http/HttpRequestUtil.java [hunk 3]
```diff
@@ -16,4 +16,20 @@
   public static void setLogEnabled(boolean enabled) {
     HTTPRequest.enableLog(enabled);
   }
+
+  /**
+   * Enable printing of the request url when a error occurred. Default value is false.
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/http/HttpRequestUtil.java]: "a error" - typo ;)

**Comment by tobrun** [comment]
@LukasPaczos, revisited the approach taken, instead of the if/else construct, I went with the `Timber.log(int severity,..` implementation instead.

**Pull request merged**